crates/ruff/src/checkers/tokens.rs

```diff
@@ -184,9 +192,39 @@ pub(crate) fn check_tokens(
     }
 
     // TD001, TD002, TD003, TD004, TD005, TD006, TD007
+    // T001, T002, T003, T004
     if enforce_todos {
+        // The TextRange of the comment, its position in comment_ranges, and the directive's
+        // TextRange
+        let mut other_directive_ranges: Vec<(TextRange, usize, TextRange)> = vec![];
+        let mut flake8_fixme_directive_ranges: Vec<(TodoDirective, TextRange)> = vec![];
+
+        // Find all TodoDirectives
+        for (i, comment_range) in indexer.comment_ranges().iter().enumerate() {
+            let comment = locator.slice(*comment_range);
+            let Some((directive, relative_offset)) = TodoDirective::from_comment(comment) else {
+                continue;
+            };
+
+            let directive_range =
+                TextRange::at(comment_range.start() + relative_offset, directive.len());
+
+            // TODO, XXX, and FIXME directives are supported by flake8_todos. flake8_fixme supports
+            // all 4 TodoDirective variants.
+            if !matches!(directive, TodoDirective::Hack) {
```

Comment:
What do you think of only using a single vec for all directives and filtering out the unsupported directives inside the rule?

Reply:
Sure, we can do that as well!